windows/lcow_runtime.go

```diff
@@ -0,0 +1,400 @@
+			Pid:       wt.pid,
+			Status:    255,
+			Timestamp: time.Now(),
+		}
+	}
+
+	wt.cleanup()
+	r.tasks.Delete(ctx, t)
+
+	r.publisher.Publish(ctx,
+		runtime.TaskDeleteEventTopic,
+		&eventsapi.TaskDelete{
+			ContainerID: wt.id,
+			Pid:         wt.pid,
+			ExitStatus:  rtExit.Status,
+			ExitedAt:    rtExit.Timestamp,
+		})
+
+	// We were never started, return failure
+	return rtExit, nil
+}
+
+func (r *LCOWRuntime) newTask(ctx context.Context, namespace, id string, spec *specs.Spec, io runtime.IO, createOpts *hcsshimopts.CreateOptions) (*task, error) {
+	var (
+		err  error
+		pset *pipeSet
+	)
+
+	if pset, err = newPipeSet(ctx, io); err != nil {
+		return nil, err
+	}
+	defer func() {
+		if err != nil {
+			pset.Close()
+		}
+	}()
+
+	var pid uint32
+	if pid, err = r.pidPool.Get(); err != nil {
+		return nil, err
+	}
+	defer func() {
+		if err != nil {
+			r.pidPool.Put(pid)
+		}
+	}()
+
+	var (
+		conf *hcsshim.ContainerConfig
+		nsid = namespace + "-" + id
+	)
+	if conf, err = newLinuxConfig(ctx, hcsshimOwner, nsid, spec); err != nil {
+		return nil, err
+	}
+
+	if err := r.CreateReadWrite(ctx, id, spec.Windows.LayerFolders[0]); err != nil {
+		return nil, err
+	}
+	conf.LayerFolderPath = r.dir(id)
+	defer func() {
+		if err != nil {
+			removeLinuxLayer(ctx, conf.LayerFolderPath)
+		}
+	}()
+
+	// TODO: remove this once we have an LCOW snapshotter
+	// Store the LayerFolder in the db so we can clean it if we die
+	if err = r.db.Update(func(tx *bolt.Tx) error {
+		s := newLayerFolderStore(tx)
+		return s.Create(nsid, conf.LayerFolderPath)
+	}); err != nil {
+		return nil, err
+	}
+	defer func() {
+		if err != nil {
+			if dbErr := r.db.Update(func(tx *bolt.Tx) error {
+				s := newLayerFolderStore(tx)
+				return s.Delete(nsid)
+			}); dbErr != nil {
+				log.G(ctx).WithField("id", id).
+					Error("failed to remove key from metadata")
+			}
+		}
+	}()
+
+	ctr, err := hcsshim.CreateContainer(nsid, conf)
+	if err != nil {
+		time.Sleep(15 * time.Second)
+		return nil, errors.Wrapf(err, "hcsshim failed to create task")
+	}
+	defer func() {
+		if err != nil {
+			ctr.Terminate()
+			ctr.Wait()
+			ctr.Close()
+		}
+	}()
+
+	if err = ctr.Start(); err != nil {
+		return nil, errors.Wrap(err, "hcsshim failed to spawn task")
+	}
+
+	t := &task{
+		id:                id,
+		namespace:         namespace,
+		pid:               pid,
+		io:                pset,
+		status:            runtime.CreatedStatus,
+		spec:              spec,
+		processes:         make(map[string]*process),
+		publisher:         r.publisher,
+		rwLayer:           conf.LayerFolderPath,
+		pidPool:           r.pidPool,
+		isWindows:         false,
+		hyperV:            true,
+		hcsContainer:      ctr,
+		terminateDuration: createOpts.TerminateDuration,
+	}
+	r.tasks.Add(ctx, t)
+
+	var rootfs []*containerdtypes.Mount
+	for _, l := range append([]string{t.rwLayer}, spec.Windows.LayerFolders...) {
+		rootfs = append(rootfs, &containerdtypes.Mount{
+			Type:   "lcow-layer",
```

Comment:
Where is this type being understood right now?

Reply:
It is not. It's just published as an event, which expects a list of Mounts for the rootfs, and the lcow layers a distinct type from any other current layers, so it didn't seem right to reuse an existing type such as `windows-layer` which is the current placeholder for Windows containers.